bot/cogs/watchchannels/watchchannel.py

```diff
@@ -0,0 +1,346 @@
+        self.api_default_params = api_default_params  # E.g., {'active': 'true', 'type': 'watch'}
+        self.log = logger  # Logger of the child cog for a correct name in the logs
+
+        # These attributes can be left as they are in the child class
+        self._consume_task = None
+        self.watched_users = defaultdict(dict)
+        self.message_queue = defaultdict(lambda: defaultdict(deque))
+        self.consumption_queue = {}
+        self.retries = 5
+        self.retry_delay = 10
+        self.channel = None
+        self.webhook = None
+        self.message_history = [None, None, 0]
+
+        self._start = self.bot.loop.create_task(self.start_watchchannel())
+
+    @property
+    def modlog(self) -> ModLog:
+        """Provides access to the ModLog cog for alert purposes."""
+        return self.bot.get_cog("ModLog")
+
+    @property
+    def consuming_messages(self) -> bool:
+        """Checks if a consumption task is currently running."""
+        if self._consume_task is None:
+            return False
+
+        if self._consume_task.done():
+            exc = self._consume_task.exception()
+            if exc:
+                self.log.exception(
+                    f"The message queue consume task has failed with:",
+                    exc_info=exc
+                )
+            return False
+
+        return True
+
+    async def start_watchchannel(self) -> None:
+        """Starts the watch channel by getting the channel, webhook, and user cache ready."""
+        await self.bot.wait_until_ready()
+
+        # After updating d.py, this block can be replaced by `fetch_channel` with a try-except
+        for attempt in range(1, self.retries+1):
+            self.channel = self.bot.get_channel(self.destination)
+            if self.channel is None:
+                if attempt < self.retries:
+                    await asyncio.sleep(self.retry_delay)
+            else:
+                break
+        else:
+            self.log.error(f"Failed to retrieve the text channel with id `{self.destination}")
+
+        # `get_webhook_info` has been renamed to `fetch_webhook` in newer versions of d.py
+        try:
+            self.webhook = await self.bot.get_webhook_info(self.webhook_id)
+        except (discord.HTTPException, discord.NotFound, discord.Forbidden):
+            self.log.exception(f"Failed to fetch webhook with id `{self.webhook_id}`")
+
+        if self.channel is None or self.webhook is None:
+            self.log.error("Failed to start the watch channel; unloading the cog.")
+
+            message = textwrap.dedent(
+                f"""
+                An error occurred while loading the text channel or webhook.
+
+                TextChannel: {"**Failed to load**" if self.channel is None else "Loaded successfully"}
+                Webhook: {"**Failed to load**" if self.webhook is None else "Loaded successfully"}
+
+                The Cog has been unloaded."""
```

Comment:
Should the closing quotes be on a new line? Is this done to prevent having to strip afterwards?

Reply:
They should indeed both be on a separate line or the same line, for consistency. It doesn't matter, as far as the embed is concerned, since the additional newlines seem to be stripped regardless. I think the version with the quotes on separate lines is more readable, as it will show you the text as it will be displayed in the embed as well.